docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/builder/DockerBuilderControlOptionRun.java

```diff
@@ -87,10 +89,16 @@ public void execute(AbstractBuild<?, ?> build) throws DockerException, IOExcepti
 
         LOGGER.info("Pulling image " + xImage);
 
-        InputStream result = client.pullImageCmd(xImage).exec();
+        PullImageResultCallback resultCallback = new PullImageResultCallback() {
+            public void onNext(PullResponseItem item) {
+                if (item.getStatus() != null && item.getProgress() == null) {
```

Comment:
progress is ok because i can see then it works, also all tools to the same. What is will be ideal is progress with `append`! But then we should know some granularity or diff to previous line... Try store curLine, nextLine, and if progress is not null, then `nextLine.replace(curline, '')` and append this tail to listener!